src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -169,6 +172,11 @@
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#RECORD_DEF">
  * RECORD_DEF</a>.
  * </li>
+ * <li>
+ * Property {@code jdkVersion} - Allow the check to be aware of the Java version.
```

Comment:
java versions are very complicated and uncertain term. We need to mention that we use major part of it, and 

https://www.java.com/releases/
we will use only first number 
```
2024-07-16 	CPU 	22.0.2 	
		"    	21.0.4 	
		"    	17.0.12 	
		"    	11.0.24 	
		"    	8u421
```

so 8, 11, ... 
we might just give link to https://en.wikipedia.org/wiki/Java_version_history , attention to `J2SE 1.4` , yes, version was `1.4`.

Reply:
https://openjdk.org/projects/jdk/

OpenJDK lists the versions with the major part only

Reply:
Ok, put this link to doc. We just need to explain what format we expect.

Should we make property as integer? To double hit on version format.

Reply:
Yep, I made it `int` and mentioned that we use major numbers only do we still need to add a link?

Reply:
I tried to search in Google what is "major version of java" and nothing found. So it is not a term, some people  can be confused.

@nrmancuso , @rnveach , what we can do to make it clear? 
May be int type of property is enough to hint what we mean

Reply:
> May be int type of property is enough to hint what we mean

I agree with this. It is pretty clear from the property type that it should be a single number only. Also, I think it is common for all Java users to mention the major part only when they asked for their Java version number.

Reply:
The term is "major version", however this is fairly convoluted by things like the JVM version, classfile versions, etc. We should use "major version"; this, coupled with the fact that this property is an int, is a clear signal that we are looking for 1.5 == 5, 22 == 22, etc.

Reply:
My thoughts at https://github.com/checkstyle/checkstyle/pull/15376#discussion_r1693589245 . We could support `1.5` as well as `5`.

Either way, I think this will require a clear description in the documentation.

Reply:
We should consider to review how maven ,Gradle , have this version. It should be variable of build system to let it be convenient.

If all build system use their own way, we can have our own versioning 5, 6, 7, 8, 9 , 10, 11, .... 22 . We just need to explain this in description

Reply:
>Should be variable of build system to let it be convenient

This makes sense, I am good with this proposal.

Maven was `1.x` until Java 9 (for the compiler plugin), gradle is the same (sourceCompatibility/targetCompatibility). We should just make this field a string, and accept 1.x for Java * and below, and X for Java 9 and above, as these build systems have. This way, if someone complains or actual usage shows us otherwise, we can support versions like 8 as well without breaking anything.

Reply:
@mahfouz72 let's do this ^, and we can create a ticket to add support for versions like `8` and discuss mapping 1.8 -> 8 there.

Reply:
done the property field  is string